pkg/provisioningv2/rke2/planner/planner.go

```diff
@@ -1248,6 +1238,57 @@ func (p *Planner) ensureRKEStateSecret(controlPlane *rkev1.RKEControlPlane, full
 	}, nil
 }
 
+func getIPAddresses(secrets corecontrollers.SecretCache, machine *capi.Machine) (string, string) {
+	internalIPAddress, internalAddressProvided := machine.Annotations[InternalAddressAnnotation]
+	ipAddress, addressProvided := machine.Annotations[AddressAnnotation]
+
+	// Unmanaged machines won't have a state secret
+	if machine.Annotations["objectset.rio.cattle.io/id"] != "unmanaged-machine" && (!internalAddressProvided || !addressProvided) {
+		secret, err := secrets.Get(machine.Spec.InfrastructureRef.Namespace, name.SafeConcatName(machine.Spec.InfrastructureRef.Name, "machine", "state"))
+		if err == nil && secret != nil && len(secret.Data["extractedConfig"]) != 0 {
+			driverConfig, err := nodeconfig.ExtractConfigJSON(base64.StdEncoding.EncodeToString(secret.Data["extractedConfig"]))
+			if err == nil && len(driverConfig) != 0 {
+				if !addressProvided {
+					ipAddress = convert.ToString(values.GetValueN(driverConfig, "Driver", "IPAddress"))
+				}
+				if !internalAddressProvided {
+					internalIPAddress = convert.ToString(values.GetValueN(driverConfig, "Driver", "PrivateIPAddress"))
+				}
+			}
+		}
+	}
+
+	return internalIPAddress, ipAddress
+}
+
+func appendToSlice(config map[string]interface{}, key string, values ...interface{}) {
+	if len(values) != 0 {
```

Comment:
This can be a one liner, `config[key] = append(convert.ToStringSlice(config[key]), values...)` similar to L693

Reply:
Done.